### PR TITLE
update: add a few more stuff to pyproject.toml

### PR DESCRIPTION
**Pull request by synchon**
* [x] description of feature/fix
* [x] tests added/passed

This PR adds the `build-backend` explicitly and also adds a few more options for `black`.

### CODE REVIEW

**Comment by synchon** [comment]
Rebased on latest `devel`.

**Comment by Midnighter** [comment]
I agree with setting the build-backend but are the includes and excludes worth the trouble of maintaining the list?

**Comment by synchon** [comment]
I have been using it like this for a few projects and seems to cause no mental overhead. Also, if you see only the last entries are added by me, except those, rest are taken from `black`'s docs.

**Comment by Midnighter** [comment]
I usually run black as `black src` directly so I've never had the need to exclude directories. And anyway only Python files are considered as far as I know. So I don't see the need personally. Does this configuration help when running black integrated in VSCode or something?

**Comment by synchon** [comment]
I don't think it does some trick for editor integration but I might be wrong. I usually follow this as the official docs put this as an example. Also, makes the lookup hierarchy a bit more explicit.

**Comment by synchon** [comment]
Rebased on latest `devel`.

File: pyproject.toml [hunk 1]
```diff
@@ -3,7 +3,30 @@ requires = [
...
+    | \.mypy_cache
+    | \.tox
+    | \.venv
+    | _build
+    | buck-out
+    | build
+    | dist
+    | documentation_builder
+    | release-notes
+    | scripts
```
Comment (Midnighter) [inline pyproject.toml]: Actually it would be nice if the Python scripts here are maintained with black.

**Review by Midnighter — APPROVED** [review]
I don't see the real benefit of adding this configuration but I also don't see a disadvantage. So fine with me.

File: pyproject.toml (hunk 1 above)
Comment (synchon) [reply pyproject.toml]: Okay I'll remove `documentation_builder` and `scripts` from the list. `dist` is usually left out and `release-notes` doesn't have any Python files.

Comment (Midnighter) [reply pyproject.toml]: Sorry, the UI was unspecific there. The comment was only on the line with `scripts`.

Comment (synchon) [reply pyproject.toml]: No worries.

**Pull request merged**